### DISCUSSION TITLE
How do I set publish time ?

### DISCUSSION THREAD

**Discussion by Thaksinna** [directus/directus#16675] [Archive]
Hi I want to set time publish from create item (publish date) .How do it.  Thank you

**Comment by josephdpurcell** [comment]
In my use case, I added a field "published_date" which the user can edit. The field is required.

I did not figure out how to set the publish time from the created time. I suggest against that for website content since there are cases where content authors want control over the time shown as published.

To set the publish time from create time (assuming publish time is a custom field you add) you could use Flows to:

1. Trigger on update of your content
2. Read that content into the flow's data chain (Read Data op)
3. Write that field (Write Data op)

**Comment by luciandavidescu** [comment]
This is a flow trying to cover all scenarios, including bulk publishing, publishing as soon as an item is created or republishing after unpublish but with the initial published date.

Make sure that date_created is **not** automatically filled on _create_ or _update_. Then, unless it's entered manually, the flow will update the db automatically on publish, date_created thus effectively becoming the published date.

1. Event-hook (action) on collection's _items.create_ & _items.update_.

2. Condition:
```
{
    "$trigger": {
        "payload": {
            "status": {
                "_eq": "published"
            }
        }
    }
}
```

3. Run script:
```
module.exports = async function(data) {
    let keys = {{$trigger}}.keys || [{{$trigger}}.key];
	return {keys,now:new Date()}
}
```

4. Update data:

- IDs:

```
{{$last.keys}}
```

- Payload: 

```
{
    "date_created": "{{$last.now}}"
}
```

- Query:

```
{
    "filter": {
        "date_created": {
            "_null": true
        }
    }
}
```

While this gets the job done, I guess it would make much more sense if it were done automatically in admin.

**Reply by DancinParrot** [reply]
Step 3's script requires a slight correction, specifically the current date should be serialized into a JSON string, hence:
```JavaScript
module.exports = async function(data) {
    let keys = {{$trigger}}.keys || [{{$trigger}}.key];
    return {
        keys,
        now: new Date().toJSON()
    }
}
```

Otherwise, null will be returned.

**Reply by sangmouse** [reply]
> This is a flow trying to cover all scenarios, including bulk publishing, publishing as soon as an item is created or republishing after unpublish but with the initial published date.
> 
> Make sure that date_created is **not** automatically filled on _create_ or _update_. Then, unless it's entered manually, the flow will update the db automatically on publish, date_created thus effectively becoming the published date.
> 
> 1. Event-hook (action) on collection's _items.create_ & _items.update_.
> 2. Condition:
> 
> ```
> {
>     "$trigger": {
>         "payload": {
>             "status": {
>                 "_eq": "published"
>             }
>         }
>     }
> }
> ```
> 
> 3. Run script:
> 
> ```
> module.exports = async function(data) {
>     let keys = {{$trigger}}.keys || [{{$trigger}}.key];
> 	return {keys,now:new Date()}
> }
> ```
> 
> 4. Update data:
> 
> * IDs:
> 
> ```
> {{$last.keys}}
> ```
> 
> * Payload:
> 
> ```
> {
>     "date_created": "{{$last.now}}"
> }
> ```
> 
> * Query:
> 
> ```
> {
>     "filter": {
>         "date_created": {
>             "_null": true
>         }
>     }
> }
> ```
> 
> While this gets the job done, I guess it would make much more sense if it were done automatically in admin.

can u share with me which docs mention this, because i refered so much document but cannot do this until i see your guide
